GiniVision/Classes/Networking/APIService.swift

```diff
@@ -170,58 +171,57 @@ final class APIService: APIServiceProtocol {
 
 extension APIService {
     
-    fileprivate func getSessionBlock(cancelationToken token: CancelationToken? = nil) -> BFContinuationBlock? {
-        return { [weak self] (task: BFTask?) -> Any! in
-            guard let `self` = self else { return nil }
-            
-            if let token = token, token.cancelled {
-                return BFTask.cancelled()
-            }
-            if task?.error != nil {
-                return self.giniSDK?.sessionManager.logIn()
+    fileprivate func getSessionBlock(cancelationToken token: BFCancellationToken? = nil)
+        -> ((BFTask<AnyObject>) -> Any?) {
```

Comment:
This return value scares me. Why didn't you continue using `BFContinuationBlock` and since it's about logging in, can't you replace the `Any` and `AnyObject` with something more specific?

Reply:
Because it breaks the compiler. For some reason it cannot infer the `ResultType`, even if you specified it.